# **STEP 1 : IMPORTER LIBRARIES**


In [ ]:
import pandas as pd

In [ ]:
import numpy as np


In [ ]:
import io

# **STEP 2 : UPLOADING LES FICHIERS CSVs**

In [ ]:
from google.colab import files


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
uploaded = files.upload()


Saving aide_alimentaire.csv to aide_alimentaire.csv
Saving dispo_alimentaire.csv to dispo_alimentaire.csv
Saving FAOSTAT_data_fr_11-1-2022.csv to FAOSTAT_data_fr_11-1-2022.csv
Saving population.csv to population.csv
Saving sous_nutrition.csv to sous_nutrition.csv


# **STEP 3: DATAFRAMES**

In [ ]:
aide_alimentaire = pd.read_csv(io.BytesIO(uploaded['aide_alimentaire.csv']))
aide_alimentaire.head(5)


Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504

In [ ]:
aide_alimentaire.shape

(1475, 4)

In [ ]:
aide_alimentaire.dtypes

Pays bénéficiaire    object
Année                 int64
Produit              object
Valeur                int64
dtype: object

In [ ]:
dispo_alimentaire = pd.read_csv(io.BytesIO(uploaded['dispo_alimentaire.csv']))
dispo_alimentaire.head(60)

Zone                   Produit   Origine  Aliments pour animaux  \
0   Afghanistan          Abats Comestible   animale                    NaN   
1   Afghanistan           Agrumes, Autres  vegetale                    NaN   
2   Afghanistan     Aliments pour enfants  vegetale                    NaN   
3   Afghanistan                    Ananas  vegetale                    NaN   
4   Afghanistan                   Bananes  vegetale                    NaN   
5   Afghanistan              Beurre, Ghee   animale                    NaN   
6   Afghanistan                     Bière  vegetale                    NaN   
7   Afghanistan                       Blé  vegetale                    NaN   
8   Afghanistan      Boissons Alcooliques  vegetale                    NaN   
9   Afghanistan                      Café  vegetale                    NaN   
10  Afghanistan        Coco (Incl Coprah)  vegetale                    NaN   
11  Afghanistan                     Crème   animale                    NaN   
12  Afghanistan          Céréales, Autres  vegetale                    NaN   
13  Afghanistan                    Dattes  vegetale                    NaN   
14  Afghanistan        Edulcorants Autres  vegetale                    NaN   
15  Afghanistan             Feve de Cacao  vegetale                    NaN   
16  Afghanistan            Fruits, Autres  vegetale                    NaN   
17  Afghanistan          Graines de coton  vegetale                    NaN   
18  Afghanistan      Graines de tournesol  vegetale                    NaN   
19  Afghanistan    Graisses Animales Crue   animale                    NaN   
20  Afghanistan   Huil Plantes Oleif Autr  vegetale                    NaN   
21  Afghanistan    Huile Graines de Coton  vegetale                    NaN   
22  Afghanistan          Huile d'Arachide  vegetale                    NaN   
23  Afghanistan             Huile d'Olive  vegetale                    NaN   
24  Afghanistan   Huile de Colza&Moutarde  vegetale                    NaN   
25  Afghanistan            Huile de Palme  vegetale                    NaN   
26  Afghanistan             Huile de Soja  vegetale                    NaN   
27  Afghanistan           Huile de Sésame  vegetale                    NaN   
28  Afghanistan        Huile de Tournesol  vegetale                    NaN   
29  Afghanistan        Lait - Excl Beurre   animale                  123.0   
30  Afghanistan           Légumes, Autres  vegetale                    NaN   
31  Afghanistan       Légumineuses Autres  vegetale                    4.0   
32  Afghanistan                      Maïs  vegetale                  200.0   
33  Afghanistan                      Miel  vegetale                    NaN   
34  Afghanistan                    Millet  vegetale                    NaN   
35  Afghanistan              Miscellanees  vegetale                    NaN   
36  Afghanistan                      Noix  vegetale                    NaN   
37  Afghanistan                     Oeufs   animale                    NaN   
38  Afghanistan                    Olives  vegetale                    NaN   
39  Afghanistan       Oranges, Mandarines  vegetale                    NaN   
40  Afghanistan                      Orge  vegetale                  360.0   
41  Afghanistan  Plantes Oleiferes, Autre  vegetale                    NaN   
42  Afghanistan        Poissons Eau Douce   animale                    NaN   
43  Afghanistan                    Poivre  vegetale                    NaN   
44  Afghanistan                    Pommes  vegetale                    NaN   
45  Afghanistan           Pommes de Terre  vegetale                    NaN   
46  Afghanistan                    Raisin  vegetale                    NaN   
47  Afghanistan          Riz (Eq Blanchi)  vegetale                    NaN   
48  Afghanistan             Sucre Eq Brut  vegetale                    NaN   
49  Afghanistan          Sucre, betterave  vegetale                    0.0   
50  Afghanistan              Sucr

In [ ]:
dispo_alimentaire.shape

(15605, 18)

In [ ]:
dispo_alimentaire.dtypes

Zone                                                              object
Produit                                                           object
Origine                                                           object
Aliments pour animaux                                            float64
Autres Utilisations                                              float64
Disponibilité alimentaire (Kcal/personne/jour)                   float64
Disponibilité alimentaire en quantité (kg/personne/an)           float64
Disponibilité de matière grasse en quantité (g/personne/jour)    float64
Disponibilité de protéines en quantité (g/personne/jour)         float64
Disponibilité intérieure                                         float64
Exportations - Quantité                                          float64
Importations - Quantité                                          float64
Nourriture                                                       float64
Pertes                                             

In [ ]:
population = pd.read_csv(io.BytesIO(uploaded['population.csv']))
population.head()

Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

In [ ]:
population.dtypes

Zone       object
Année       int64
Valeur    float64
dtype: object

In [ ]:
population['Année'] = population['Année'].apply(str)

In [ ]:
population['Année'].astype(object)

0       2013
1       2014
2       2015
3       2016
4       2017
        ... 
1411    2014
1412    2015
1413    2016
1414    2017
1415    2018
Name: Année, Length: 1416, dtype: object

In [ ]:
population.dtypes

Zone       object
Année      object
Valeur    float64
dtype: object

In [ ]:
sous_nutrition = pd.read_csv(io.BytesIO(uploaded['sous_nutrition.csv']))
sous_nutrition.head()

Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5

In [ ]:
sous_nutrition.dtypes

Zone      object
Année     object
Valeur    object
dtype: object

In [ ]:
sous_nutrition['Valeur'] = sous_nutrition['Valeur'].str.replace('<0.1','0').astype(float)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: The default value of regex will change from True to False in a future version.
  """Entry point for launching an IPython kernel.


In [ ]:
sous_nutrition.dtypes

Zone       object
Année      object
Valeur    float64
dtype: object

In [ ]:
#modifier le nom de la serie "Valeur" par "Population_sous_alimentée"
sous_nutrition = sous_nutrition.rename(columns={"Valeur": "Population_sous_alimentée"})
sous_nutrition.head()

Zone      Année  Population_sous_alimentée
0  Afghanistan  2012-2014                        8.6
1  Afghanistan  2013-2015                        8.8
2  Afghanistan  2014-2016                        8.9
3  Afghanistan  2015-2017                        9.7
4  Afghanistan  2016-2018                       10.5

In [ ]:
sous_nutrition.dtypes

Zone                          object
Année                         object
Population_sous_alimentée    float64
dtype: object

In [ ]:
sous_nutrition['Année']

0       2012-2014
1       2013-2015
2       2014-2016
3       2015-2017
4       2016-2018
          ...    
1213    2013-2015
1214    2014-2016
1215    2015-2017
1216    2016-2018
1217    2017-2019
Name: Année, Length: 1218, dtype: object

In [ ]:
sous_nutrition['Population_sous_alimentée'] = pd.to_numeric(sous_nutrition['Population_sous_alimentée'])

In [ ]:
sous_nutrition.dtypes

Zone                          object
Année                         object
Population_sous_alimentée    float64
dtype: object

# **STEP 4 : QUESTIONS MARC**

## **4.1 La sous-nutrition en 2017**

### **4.1.1 La population mondiale en 2017**

In [ ]:
#Le nombre total de la population en 2017
pop_2017=round(population['Valeur'][population['Année']=='2017'].sum())*1000
pop_2017

7548134000

**7,54 milliards** de personnes en 2017

### **4.1.2 Le nombre total de la population en état de sous-nutrition**

In [ ]:
#Le nombre total de la population en état de sous-nutrition
pop_sous_nutrition2017 = (sous_nutrition['Population_sous_alimentée'][sous_nutrition['Année'] == '2016-2018'].sum())
pop_sous_nutrition2017=(pop_sous_nutrition2017)*1000
pop_sous_nutrition2017

535700.0

**535,70 Millions de personnes** dans le monde étaient en état de sous-nutrition en 2017

In [ ]:
#La proportion de la population mondiale en état de sous-nutrition
prop_personne_sous_nutrition2017=(pop_sous_nutrition2017*100)/(pop_2017)
prop_personne_sous_nutrition2017*1000

7.0971183076506055

Soit **7,09%** de la population mondiale.

## **4.2 La disponibilité alimentaire**

### **4.2.1 Le nombre théorique de personnes qui pourraient être nourries en 2017 considerant l'apport calorique disponible au monde**




In [ ]:
dispo_alimentaire

In [ ]:
population

In [ ]:
dispo_alimentaire

In [ ]:
population

In [ ]:
#Attention à la relation kcal par personne sur le tableau dispo_alimentaire. Cette serie
#ne nous permet pas de connaitre le nombre reel de la disponibilite calorique dans un
#pays. Pour cela, il faut, d'abord, réaliser un groupby par pays sur le df dispo_alimentaire
#On a ainsi un nouveau df "dispo_alimentaire_2017"
dispo_alimentaire_2017=dispo_alimentaire[['Zone','Produit','Disponibilité alimentaire (Kcal/personne/jour)']]
dispo_alimentaire_2017=dispo_alimentaire_2017.groupby(by='Zone').agg({'Disponibilité alimentaire (Kcal/personne/jour)':sum}).sort_values(by='Disponibilité alimentaire (Kcal/personne/jour)', ascending=False)
dispo_alimentaire_2017

In [ ]:
#Ensuite on filtre les données sur le df population on choisissant l'année cicle 2017
#On a un nouveau df "dispo_alimentaire_pop_2017"
dispo_alimentaire_pop_2017=(population[(population['Année']=='2017')])
dispo_alimentaire_pop_2017

Zone Année     Valeur
4                                Afghanistan  2017  36296.113
10                            Afrique du Sud  2017  57009.756
16                                   Albanie  2017   2884.169
22                                   Algérie  2017  41389.189
28                                 Allemagne  2017  82658.409
...                                      ...   ...        ...
1390  Venezuela (République bolivarienne du)  2017  29402.484
1396                                Viet Nam  2017  94600.648
1402                                   Yémen  2017  27834.819
1408                                  Zambie  2017  16853.599
1414                                Zimbabwe  2017  14236.595

[236 rows x 3 columns]

In [ ]:
#On fait une jonction des dfs afin d'avoir sur un seul tableau l'info de calories
#produite par personne et la population total de chaque pays
#Un nouveau df apparait: merge_dispo_pop_2017
merge_dispo_pop_2017=dispo_alimentaire_pop_2017.merge(dispo_alimentaire_2017, on = ['Zone'])
merge_dispo_pop_2017

In [ ]:
#Maintenant on multiplie la population d'un pays avec la disponibilite alimentaire
#calorique par personne.
#La nouvelle serie "Total" affichera la disponibilite journalière total cette fois-ci tentant en
#compte la population de chaque pays en 2017
merge_dispo_pop_2017['Total'] = merge_dispo_pop_2017['Disponibilité alimentaire (Kcal/personne/jour)']*merge_dispo_pop_2017['Valeur']
merge_dispo_pop_2017

Zone Année     Valeur  \
0                               Afghanistan  2017  36296.113   
1                            Afrique du Sud  2017  57009.756   
2                                   Albanie  2017   2884.169   
3                                   Algérie  2017  41389.189   
4                                 Allemagne  2017  82658.409   
..                                      ...   ...        ...   
167  Venezuela (République bolivarienne du)  2017  29402.484   
168                                Viet Nam  2017  94600.648   
169                                   Yémen  2017  27834.819   
170                                  Zambie  2017  16853.599   
171                                Zimbabwe  2017  14236.595   

     Disponibilité alimentaire (Kcal/personne/jour)         Total  
0                                            2087.0  7.574999e+07  
1                                            3020.0  1.721695e+08  
2                                            3188.0  9.194731e+06  
3                                            3293.0  1.362946e+08  
4                                            3503.0  2.895524e+08  
..                                              ...           ...  
167                                          2633.0  7.741674e+07  
168                                          2744.0  2.595842e+08  
169                                          2217.0  6.170979e+07  
170                                          1924.0  3.242632e+07  
171                                          2113.0  3.008193e+07  

[172 rows x 5 columns]

In [ ]:
#On fait la somme de la série totale afin de connaitre le volume reel produit par jour
dispo_kcal_jour=round(merge_dispo_pop_2017['Total'].sum())
dispo_kcal_jour


20918984627

En 2017 le monde a produit environs **21 milliards de calories par jour**

In [ ]:
#On trouve le volume annuel
dispo_kcal_an=round(dispo_kcal_jour)*365
dispo_kcal_an

7635429388855

Cela represente environs **7 trillions de calories produite en 2017**

In [ ]:
#On divisant le volume annuel de calories produite en 2017 on peut, enfin, trouver le
#nombre de personnes qui auraient pu, en théorie, être nourries
#On considere qu'un être humain a besoin de 912500kcal par an (ou 2500kcal par jour) pour vivre, selon l'OMS
nombre_personnes_nourries_kcal=round(dispo_kcal_an/912500)*1000
nombre_personnes_nourries_kcal

8367594000

Le nombre total de calories produites en 2017 aurait pu nourrir **8,3 milliards de personne**

In [ ]:
#Maintenant on calcule la proportion de personne dans une échelle mondiale
prop_nombre_personnes_nourries=(nombre_personnes_nourries_kcal*100/pop_2017)
prop_nombre_personnes_nourries

110.85645803320396

Le chiffre au-dessus represente **110% de la population mondiale**

3.1.1 Disponibilité alimentaire en **proteines**

### **4.2.2 Le nombre théorique de personnes qui pourraient être nourries en 2017 considerant le volume total de proteine disponible au monde**

In [ ]:
dispo_alimentaire_2017_proteine=dispo_alimentaire[['Zone','Produit','Disponibilité de protéines en quantité (g/personne/jour)']]
dispo_alimentaire_2017_proteine=dispo_alimentaire_2017_proteine.groupby(by='Zone').agg({'Disponibilité de protéines en quantité (g/personne/jour)':sum}).sort_values(by='Disponibilité de protéines en quantité (g/personne/jour)', ascending=False)
dispo_alimentaire_2017_proteine

In [ ]:
merge_dispo_pop_2017_proteine=dispo_alimentaire_pop_2017.merge(dispo_alimentaire_2017_proteine, on = ['Zone'])
merge_dispo_pop_2017_proteine

In [ ]:
merge_dispo_pop_2017_proteine['Total'] = merge_dispo_pop_2017_proteine['Disponibilité de protéines en quantité (g/personne/jour)']*merge_dispo_pop_2017_proteine['Valeur']
merge_dispo_pop_2017_proteine

In [ ]:
dispo_proteine_jour=round(merge_dispo_pop_2017_proteine['Total'].sum())
dispo_proteine_jour

587063674

En 2017 le monde a produit **587 millions de grammes de proteines par jour** (ou 587t de proteine)

In [ ]:
dispo_proteine_an=round(dispo_proteine_jour)*365
dispo_proteine_an

214278241010

Cela veut dire qu'en 2017, la production totale de proteine a atteint **214 milliards de grammes de proteine** (ou 214 mille tonnes de proteine)

In [ ]:
#Nombre de personne qui pourrait être nourris en 2017 en respectant un régime avec 44g de proteine par jour ou 15840g par an (recommendation de l'OMS)
nombre_personnes_nourries_proteine=round(dispo_proteine_an/15840)*1000
nombre_personnes_nourries_proteine

13527667000

**13 milliards de personnes** aurait pu être nourries en 2017 grace au volume de proteine produit en 2017

In [ ]:
prop_nombre_personnes_nourries_proteine=(nombre_personnes_nourries_proteine*100/pop_2017)
prop_nombre_personnes_nourries_proteine

179.2186916660462

ce qui represente **179% de la population mondiale**

### **4.2.3 Le nombre théorique de personnes qui pourraient être nourries en 2017 considerant la disponibilite de produits d'origine vegetale au monde**

4.2.3.1 Considerant le besoin humain en apport **calorique** 

In [ ]:
dispo_alimentaire.head()

In [ ]:
dispo_alimentaire_2017_vegetale_kcal=dispo_alimentaire[['Zone','Produit', 'Origine','Disponibilité alimentaire (Kcal/personne/jour)']]
dispo_alimentaire_2017_vegetale_kcal=dispo_alimentaire_2017_vegetale_kcal[dispo_alimentaire_2017_vegetale_kcal['Origine']=='vegetale']
dispo_alimentaire_2017_vegetale_kcal

In [ ]:
merge_dispo_pop_2017_veg_kcal=dispo_alimentaire_pop_2017.merge(dispo_alimentaire_2017_vegetale_kcal, on = ['Zone'])
merge_dispo_pop_2017_veg_kcal

In [ ]:
merge_dispo_pop_2017_veg_kcal['Total'] = merge_dispo_pop_2017_veg_kcal['Disponibilité alimentaire (Kcal/personne/jour)']*merge_dispo_pop_2017_veg_kcal['Valeur']
merge_dispo_pop_2017_veg_kcal

In [ ]:
dispo_kcal_jour_veg=round(merge_dispo_pop_2017_veg_kcal['Total'].sum())
dispo_kcal_jour_veg

17260764212

En 2017 la production d'aliment d'origine vegetale a été responsable pour la production de environs **17 milliards de calories par jour**

In [ ]:
dispo_kcal_an_veg=round(dispo_kcal_jour_veg)*365
dispo_kcal_an_veg

6300178937380

ce qui veut dire une production totale d'environs **6 trillions de calories en 2017**

In [ ]:
nombre_personnes_nourries_kcal_veg=round(dispo_kcal_an_veg/912500)*1000
nombre_personnes_nourries_kcal_veg

6904306000

Le nombre total de calories produites d'origine vegetale en 2017 aurait pu nourrir près de **7 milliards de personnes**

In [ ]:
prop_nombre_personnes_nourries_veg=(nombre_personnes_nourries_kcal_veg*100/pop_2017)
prop_nombre_personnes_nourries_veg

91.47036870304635

Cela represente **91% de la population mondiale**

4.2.3.2 Considerant le besoin humain en **proteine**

In [ ]:
dispo_alimentaire_2017_vegetale_proteine=dispo_alimentaire[['Zone','Produit', 'Origine','Disponibilité de protéines en quantité (g/personne/jour)']]
dispo_alimentaire_2017_vegetale_proteine=dispo_alimentaire_2017_vegetale_proteine[dispo_alimentaire_2017_vegetale_proteine['Origine']=='vegetale']
dispo_alimentaire_2017_vegetale_proteine

In [ ]:
merge_dispo_pop_2017_veg_proteine=dispo_alimentaire_pop_2017.merge(dispo_alimentaire_2017_vegetale_proteine, on = ['Zone'])
merge_dispo_pop_2017_veg_proteine

In [ ]:
merge_dispo_pop_2017_veg_proteine['Total'] = merge_dispo_pop_2017_veg_proteine['Disponibilité de protéines en quantité (g/personne/jour)']*merge_dispo_pop_2017_veg_proteine['Valeur']
merge_dispo_pop_2017_veg_proteine

In [ ]:
dispo_proteine_jour_veg=round(merge_dispo_pop_2017_veg_proteine['Total'].sum())
dispo_proteine_jour_veg

358050057

En 2017 le monde a produit **358 millions de grammes de proteines par jour** (ou 358t de proteine)

In [ ]:
dispo_proteine_an_veg=round(dispo_proteine_jour_veg)*365
dispo_proteine_an_veg

130688270805

ce qui veut dire une production totale d'environs **130 milliards de proteines en 2017** (ou 130 mille tonnes de proteine)

In [ ]:
nombre_personnes_nourries_proteine_vegetale=round(dispo_proteine_an_veg/15840)*1000
nombre_personnes_nourries_proteine_vegetale

8250522000

Le nombre total de proteine d'origine vegetale produites en 2017 aurait pu nourrir **8,2 milliards de personnes**

In [ ]:
prop_nombre_personnes_nourries_proteine_vegetale=(nombre_personnes_nourries_proteine_vegetale*100/pop_2017)
prop_nombre_personnes_nourries_proteine_vegetale

109.3054521819565

Cela represente **109% de la population mondiale**

4.1.2.3.3 Les **produits** d'origine vegetale

In [ ]:
dispo_alimentaire_2017_veget=dispo_alimentaire[['Zone','Produit', 'Origine','Production']]
dispo_alimentaire_2017_veget=dispo_alimentaire_2017_veget[dispo_alimentaire_2017_veget['Origine']=='vegetale']
dispo_alimentaire_2017_veget

In [ ]:
les_produits_vegetale_2017=dispo_alimentaire_2017_veget.groupby('Produit').sum().sort_values(['Production'], ascending=False)
les_produits_vegetale_2017.head(30)

Production
Produit                           
Sucre, canne             1879812.0
Maïs                     1017029.0
Légumes, Autres           882664.0
Blé                       708443.0
Riz (Eq Blanchi)          495447.0
Pommes de Terre           374223.0
Soja                      278361.0
Manioc                    255002.0
Sucre, betterave          247026.0
Fruits, Autres            203580.0
Bière                     188586.0
Sucre Eq Brut             179206.0
Tomates                   162334.0
Orge                      142462.0
Bananes                   105747.0
Patates douces            100470.0
Oranges, Mandarines        97076.0
Oignons                    84614.0
Pommes                     82165.0
Raisin                     77885.0
Graines Colza/Moutarde     73392.0
Ignames                    62563.0
Sorgho                     62116.0
Coco (Incl Coprah)         60790.0
Huile de Palme             53612.0
Graines de coton           44899.0
Graines de tournesol       44607.0
Huile de Soja              42635.0
Légumineuses Autres        38851.0
Arachides Decortiquees     31920.0

### **4.2.4 Et si l'on considère la disponibilité de produits d'origine animale au monde, combien de personnes auraient pu être nourries en 2017?**

4.2.4.1 Considerant le besoin humain en **calories**

In [ ]:
dispo_alimentaire

In [ ]:
dispo_alimentaire_2017_animal_kcal=dispo_alimentaire[['Zone','Produit', 'Origine','Disponibilité alimentaire (Kcal/personne/jour)']]
dispo_alimentaire_2017_animal_kcal=dispo_alimentaire_2017_animal_kcal[dispo_alimentaire_2017_animal_kcal['Origine']=='animale']
dispo_alimentaire_2017_animal_kcal

In [ ]:
merge_dispo_pop_2017_animal_kcal=dispo_alimentaire_pop_2017.merge(dispo_alimentaire_2017_animal_kcal, on = ['Zone'])
merge_dispo_pop_2017_animal_kcal

In [ ]:
merge_dispo_pop_2017_animal_kcal['Total'] = merge_dispo_pop_2017_animal_kcal['Disponibilité alimentaire (Kcal/personne/jour)']*merge_dispo_pop_2017_animal_kcal['Valeur']
merge_dispo_pop_2017_animal_kcal

In [ ]:
dispo_kcal_jour_animal=round(merge_dispo_pop_2017_animal_kcal['Total'].sum())
dispo_kcal_jour_animal

3658220416

In [ ]:
dispo_kcal_an_animal=round(dispo_kcal_jour_animal)*365
dispo_kcal_an_animal

1335250451840

In [ ]:
nombre_personnes_nourries_kcal_animal=round(dispo_kcal_an_animal/912500)*1000
nombre_personnes_nourries_kcal_animal

1463288000

Les aliments **d'origine animale** auraint pu nourrir **1,46 milliards de personnes en 2017**

In [ ]:
prop_nombre_personnes_nourries_animal=(nombre_personnes_nourries_kcal_animal*100/pop_2017)
prop_nombre_personnes_nourries_animal

19.386089330157624

Cela represente **19% de la population mondiale**

4.2.4.2 Considerant le besoin humain en **proteine**

In [ ]:
dispo_alimentaire_2017_animal_proteine=dispo_alimentaire[['Zone','Produit', 'Origine','Disponibilité de protéines en quantité (g/personne/jour)']]
dispo_alimentaire_2017_animal_proteine=dispo_alimentaire_2017_animal_proteine[dispo_alimentaire_2017_animal_proteine['Origine']=='animale']
dispo_alimentaire_2017_animal_proteine

In [ ]:
merge_dispo_pop_2017_animal_proteine=dispo_alimentaire_pop_2017.merge(dispo_alimentaire_2017_animal_proteine, on = ['Zone'])
merge_dispo_pop_2017_animal_proteine

In [ ]:
merge_dispo_pop_2017_animal_proteine['Total'] = merge_dispo_pop_2017_animal_proteine['Disponibilité de protéines en quantité (g/personne/jour)']*merge_dispo_pop_2017_animal_proteine['Valeur']
merge_dispo_pop_2017_animal_proteine

In [ ]:
dispo_proteine_jour_animal=round(merge_dispo_pop_2017_animal_proteine['Total'].sum())
dispo_proteine_jour_animal

229013617

In [ ]:
dispo_proteine_an_animal=round(dispo_proteine_jour_animal)*365
dispo_proteine_an_animal

83589970205

In [ ]:
nombre_personnes_nourries_proteine_animal=round(dispo_proteine_an_animal/15840)*1000
nombre_personnes_nourries_proteine_animal

5277145000

Le nombre total de proteine d'origine animale produite en 2017 aurait pu nourrir **5,2 milliards de personnes**

In [ ]:
prop_nombre_personnes_nourries_proteine_animal=(nombre_personnes_nourries_proteine_animal*100/pop_2017)
prop_nombre_personnes_nourries_proteine_animal

69.91323948408971

Cela represente près de **70% de la population mondiale**

4.2.4.3 **Les 20 plus grands éléveurs de bétail au monde**

In [ ]:
dispo_alimentaire

In [ ]:
dispo_production_2017_animal=dispo_alimentaire[['Zone','Produit', 'Origine', 'Production']]
dispo_production_2017_animal=dispo_production_2017_animal[dispo_production_2017_animal['Origine']=='animale']
dispo_production_2017_animal

In [ ]:
les_eleveurs_betail_2017=dispo_production_2017_animal.groupby('Zone').sum().sort_values(['Production'], ascending=False)
les_eleveurs_betail_2017.head(20)

Production
Zone                             
Chine, continentale      230681.0
Inde                     158226.0
États-Unis d'Amérique    154474.0
Brésil                    65202.0
Fédération de Russie      47749.0
Pakistan                  44550.0
Allemagne                 43621.0
France                    33525.0
Turquie                   23196.0
Nouvelle-Zélande          22234.0
Mexique                   22213.0
Indonésie                 20382.0
Royaume-Uni               19783.0
Japon                     18753.0
Argentine                 18743.0
Pologne                   18413.0
Italie                    17055.0
Pays-Bas                  16943.0
Espagne                   16118.0
Australie                 15685.0

4.2.4.4 Les **produits** d'origine animale

In [ ]:
les_produits_eleveurs_betail_2017=dispo_production_2017_animal.groupby('Produit').sum().sort_values(['Production'], ascending=False)
les_produits_eleveurs_betail_2017.head(30)

Production
Produit                             
Lait - Excl Beurre          753035.0
Viande de Suides            113074.0
Viande de Volailles         108522.0
Oeufs                        73791.0
Viande de Bovins             67546.0
Poissons Eau Douce           52329.0
Poissons Pelagiques          36632.0
Plantes Aquatiques           23772.0
Graisses Animales Crue       23276.0
Perciform                    20768.0
Abats Comestible             18591.0
Mollusques, Autres           17825.0
Viande d'Ovins/Caprins       13639.0
Crustacés                    12603.0
Poissons Marins, Autres      10784.0
Beurre, Ghee                  9763.0
Viande, Autre                 6360.0
Cephalopodes                  3884.0
Crème                         3394.0
Animaux Aquatiques Autre      1427.0
Huiles de Poissons            1025.0
Huiles de Foie de Poisso        14.0
Viande de Anim Aquatiq           0.0

### **4.2.5 L'utilisation de la disponibilité alimentaire attribuée à l'alimentation animale**

In [ ]:
dispo_alimentaire_animaux=dispo_alimentaire['Aliments pour animaux'].sum()
dispo_alimentaire_animaux*1000

1304245000.0

Près de **1 milliard de tonnes** d'aliment ont été attribuée à l'alimentation animale

In [ ]:
dispo_alimentaire_interieure_tt=dispo_alimentaire['Disponibilité intérieure'].sum()
dispo_alimentaire_interieure_tt

9848994.0

In [ ]:
#La proportion destinée à l'alimentation animale
prop_alimentation_animaux=round(dispo_alimentaire_animaux/dispo_alimentaire_interieure_tt*100, 2)
prop_alimentation_animaux

13.24

**13%** de tout aliment produit en 2017 a été destinés à nourrir les animaux

### **4.2.6 L'utilisation de la disponibilité alimentaire attribuée à l'alimentation humaine**

In [ ]:
dispo_alimentaire.head(60)

In [ ]:
#On réalise la somme de la série 'Nourriture' afin de connaitre le volume d'aliment destinée à l'alimentation humaine
dispo_alimentaire_humains=dispo_alimentaire['Nourriture'].sum()*1000
dispo_alimentaire_humains

4876258000.0

*Près* de **5 milliards de tonnes** d'aliment ont été attribuée à l'alimentation humaine

In [ ]:
#Avant de calculer la proportion d'aliment destinée à l'alimentation humaine, il faut connaitre aussi le total de la
#disponibilité intérieure (la disponibilite locale de chaque pays)
dispo_alimentaire_interieure_tt=dispo_alimentaire['Disponibilité intérieure'].sum()*1000
dispo_alimentaire_interieure_tt

9848994000.0

En 2017 le monde disposait de presque **10 milliards de tonnes d'aliment**

In [ ]:
#La proportion destinée à l'alimentation humaine
prop_humains=round(dispo_alimentaire_humains/dispo_alimentaire_interieure_tt*100, 2)
prop_humains

49.51

**49%** de tout aliment produit en 2017 a été destiné à nourrir les être humains

### **4.1.2.6 Le gaspillage par rapport la disponibilité alimentaire au monde**

In [ ]:
dispo_alimentaire_pertes=dispo_alimentaire['Pertes'].sum()
dispo_alimentaire_pertes*1000
dispo_alimentaire_pertes

453698.0

**453 millions de tonnes d'aliment** ont été gaspillé en 2017

In [ ]:
#Proportion perdue par rapport la disponibilite alimentaire
prop_pertes=(453698000.0/9848994000.0*100)
prop_pertes

4.606541541196999

Cela represente **4.6%** de la production totale d'aliment en 2017

# **STEP 5 : QUESTIONS JULIEN**

## **5.1 Calcule des variables**

**Disponibilité intérieur** = Production + Importations - Exportations + Evolutions des stocks ;

**Nourriture** = Disponibilité intérieure - Semences - Pertes - Aliments pour animaux - Autre utilisation - Traitement ;


## **5.2 Analyse sur l'utilisation de céréales**


In [ ]:
#création df "céréales"
céréales = pd.read_csv(io.BytesIO(uploaded['FAOSTAT_data_fr_11-1-2022.csv']))
céréales.head()


### **5.2.1 Les céréales les plus disponibles au monde**

In [ ]:
table_dispo_céréales = pd.pivot_table(céréales, values='Valeur', index=['Produit'],
                    columns=['Élément'], aggfunc=np.sum)
table_dispo_céréales.sort_values(by=['Disponibilité intérieure'],ascending=False)


Élément             Alimentation pour touristes  Aliments pour animaux  \
Produit                                                                  
Maïs et produits                           34.0               633351.0   
Riz et produits                           561.0                40284.0   
Blé et produits                           756.0               133469.0   
Orge et produits                            1.0                91158.0   
Sorgho et produits                          1.0                25668.0   
Céréales, Autres                            1.0                19052.0   
Millet et produits                          0.0                 4012.0   
Avoine                                      3.0                16489.0   
Seigle et produits                          0.0                 8328.0   

Élément             Autres utilisations (non alimentaire)  \
Produit                                                     
Maïs et produits                                 218981.0   
Riz et produits                                   20684.0   
Blé et produits                                   20653.0   
Orge et produits                                   2281.0   
Sorgho et produits                                 2273.0   
Céréales, Autres                                    776.0   
Millet et produits                                   48.0   
Avoine                                               75.0   
Seigle et produits                                   55.0   

Élément             Disponibilité alimentaire (Kcal/personne/jour)  \
Produit                                                              
Maïs et produits                                           31931.0   
Riz et produits                                            54811.0   
Blé et produits                                           100881.0   
Orge et produits                                            1996.0   
Sorgho et produits                                          5018.0   
Céréales, Autres                                            1920.0   
Millet et produits                                          3780.0   
Avoine                                                       962.0   
Seigle et produits                                          1849.0   

Élément             Disponibilité alimentaire en quantité (kg/personne/an)  \
Produit                                                                      
Maïs et produits                                              3796.97        
Riz et produits                                               7830.83        
Blé et produits                                              12639.85        
Orge et produits                                               266.23        
Sorgho et produits                                             614.62        
Céréales, Autres                                               208.54        
Millet et produits                                             487.08        
Avoine                                                         140.60        
Seigle et produits                                             246.30        

Élément             Disponibilité de matière grasse en quantité (g/personne/jour)  \
Produit                                                                             
Maïs et produits                                               287.75               
Riz et produits                                                125.81               
Blé et produits                                                561.53               
Orge et produits                                                 7.82               
Sorgho et produits                                              47.56               
Céréales, Autres                                                 6.98               
Millet et produits                                              35.06               
Avoine                                                          16.49               
Seigle et produits               

In [ ]:
#Production totale de céréales en 2013
table_dispo_céréales['Disponibilité intérieure'].sum()*1000

3133978000.0

### **5.2.2 Les céréales les plus utilisés pour l'alimentation animale**

In [ ]:
table_dispo_céréales_pour_animaux = pd.pivot_table(table_dispo_céréales, values='Aliments pour animaux', index=['Produit'], aggfunc=np.sum)
table_dispo_céréales_pour_animaux.sort_values(by=['Aliments pour animaux'],ascending=False)

Élément             Aliments pour animaux
Produit                                  
Maïs et produits                 633351.0
Blé et produits                  133469.0
Orge et produits                  91158.0
Riz et produits                   40284.0
Sorgho et produits                25668.0
Céréales, Autres                  19052.0
Avoine                            16489.0
Seigle et produits                 8328.0
Millet et produits                 4012.0

### **5.2.3 Les céréales les plus utilisés pour l'alimentation humaine**

In [ ]:
table_dispo_céréales_pour_humains = pd.pivot_table(table_dispo_céréales, values='Nourriture', index=['Produit'], aggfunc=np.sum)
table_dispo_céréales_pour_humains.sort_values(by=['Nourriture'],ascending=False)

Élément             Nourriture
Produit                       
Riz et produits       753184.0
Blé et produits       568296.0
Maïs et produits      138113.0
Sorgho et produits     27554.0
Millet et produits     21200.0
Céréales, Autres        7491.0
Orge et produits        6617.0
Seigle et produits      5218.0
Avoine                  4528.0

## **5.3 La manioc en Thailand**

### **5.3.1 Les 10 plus grands exportateurs de manioc au monde**

In [ ]:
#creation du df dispo_pays_manioc pour obtenir la liste des grands producteurs du manioc
pays_producteurs_manioc=(dispo_alimentaire[(dispo_alimentaire['Produit']=='Manioc')&(dispo_alimentaire['Exportations - Quantité']!=0)])
pays_producteurs_manioc=pays_producteurs_manioc.dropna(subset=['Exportations - Quantité'])
pays_producteurs_manioc=pays_producteurs_manioc[['Zone', 'Exportations - Quantité']].sort_values(by='Exportations - Quantité', ascending=False)
pays_producteurs_manioc.head(11)

Zone  Exportations - Quantité
13809                  Thaïlande                  25214.0
14743                   Viet Nam                   8973.0
6334                   Indonésie                    820.0
2665                    Cambodge                    254.0
10752                   Paraguay                    164.0
3698                  Costa Rica                    123.0
3228   Chine, Taiwan Province de                     82.0
3043    Chine - RAS de Hong-Kong                     80.0
10386                    Ouganda                     47.0
3323         Chine, continentale                     41.0
2129                      Brésil                     38.0

In [ ]:
#le nombre des pays producteur du manioc
pays_producteurs_manioc['Zone'].nunique()

44

### **5.3.2 Le volume total du manioc produit en Thailande par rapport la prodution mondiale**

In [ ]:
dispo_alimentaire

In [ ]:
#creation du df_manioc ayant le df dispo_alimentaire comme reference le df dispo_alimentaire
#production totale du manioc en Thailand
dispo_manioc_production=(dispo_alimentaire[(dispo_alimentaire['Zone']=='Thaïlande') & (dispo_alimentaire['Produit']=='Manioc')])
dispo_manioc_production=(dispo_manioc_production['Production']).sum()
dispo_manioc_production

30228.0

En 2017, la Thailande a produit **30 mille tonnes de manioc**

In [ ]:
#production mondiale du manioc
production_manioc_monde=(dispo_alimentaire['Produit']=='Manioc').sum()*1000
production_manioc_monde

167000

La production mondiale du manioc a atteint **167 mille tonnes**

In [ ]:
#calcule de la proportion
prop_manioc_prod_thai_monde=(dispo_manioc_production*100)/(production_manioc_monde)
prop_manioc_prod_thai_monde

18.10059880239521

**18% du manioc produite** en 2017 au monde est venu de la Thailande

### **5.3.3 Le volume de manioc exporté par le pays**



In [ ]:
#creation du df dispo_manioc_export afin de connaitre le volume exporte par la Thailande
dispo_manioc_export_thailande=pays_producteurs_manioc[['Zone', 'Exportations - Quantité']][(pays_producteurs_manioc['Zone']=='Thaïlande')]
dispo_manioc_export_thailande

Zone  Exportations - Quantité
13809  Thaïlande                  25214.0

La Thailande a exporté en 2017 **25 mille tonnes de sa production**

In [ ]:
prop_mandioc_export_thailande=(dispo_manioc_export_thailande['Exportations - Quantité']*100)/(dispo_manioc_production)
prop_mandioc_export_thailande

13809    83.41273
Name: Exportations - Quantité, dtype: float64

ce qui represente **83% de sa production**

### **5.3.4 La sous-nutrition en Thailande**

**5.3.4.1 La population Thai**

In [ ]:
#la population de la Thailande
pop_thailande=population[['Zone','Valeur']][(population['Zone']=='Thaïlande')&(population['Année']=='2013')]
pop_thailande

Zone     Valeur
1308  Thaïlande  68144.518

La population de la Thailande est de **68 millions de personnes**

In [ ]:
#le nombre de la population en Thailande consideree sous-alimente
sous_nutrition_thailande=sous_nutrition[['Zone','Population_sous_alimentée']][(sous_nutrition['Zone'] == 'Thaïlande')&(sous_nutrition['Année']=='2012-2014')]
sous_nutrition_thailande

Zone  Population_sous_alimentée
1110  Thaïlande                        6.2

et environs **6 millions de personnes** sont en état de sous-nutrition, ce qui represente **10%** de la population totale

# **STEP 6: QUESTIONS MELANIE**

## **6.1 Les pays pour lesquels la proportion de personnes sous-alimentées est la plus forte en 2017**


In [ ]:
sous_nutrition_2017=(sous_nutrition[(sous_nutrition['Année']=='2016-2018')])
sous_nutrition_2017=sous_nutrition_2017.dropna(subset=['Population_sous_alimentée'])
sous_nutrition_2017=sous_nutrition_2017[['Zone', 'Population_sous_alimentée']].sort_values(by='Population_sous_alimentée', ascending=False)
sous_nutrition_2017.head(20)

In [ ]:
merge_pays_sous_nutrition_plus=dispo_alimentaire_pop_2017.merge(sous_nutrition_2017, on = ['Zone'])
merge_pays_sous_nutrition_plus

In [ ]:
merge_pays_sous_nutrition_plus['Ratio sous_nutrition']=merge_pays_sous_nutrition_plus['Population_sous_alimentée']*100/merge_pays_sous_nutrition_plus['Valeur']
merge_pays_sous_nutrition_plus=merge_pays_sous_nutrition_plus.sort_values(by='Ratio sous_nutrition', ascending=False)
merge_pays_sous_nutrition_plus.head(20)

Zone Année      Valeur  \
41                                        Haïti  2017   10982.366   
82   République populaire démocratique de Corée  2017   25429.825   
57                                   Madagascar  2017   25570.512   
55                                      Libéria  2017    4702.226   
53                                      Lesotho  2017    2091.534   
94                                        Tchad  2017   15016.753   
84                                       Rwanda  2017   11980.961   
66                                   Mozambique  2017   28649.018   
96                                  Timor-Leste  2017    1243.258   
0                                   Afghanistan  2017   36296.113   
24                                        Congo  2017    5110.695   
102      Venezuela (République bolivarienne du)  2017   29402.484   
89                                 Sierra Leone  2017    7488.423   
83                  République-Unie de Tanzanie  2017   54660.339   
47                                         Iraq  2017   37552.781   
50                                        Kenya  2017   50221.142   
13                                     Botswana  2017    2205.080   
97                                         Togo  2017    7698.474   
26                                Côte d'Ivoire  2017   24437.470   
33                                     Éthiopie  2017  106399.924   

     Population_sous_alimentée  Ratio sous_nutrition  
41                         5.3              0.048259  
82                        12.0              0.047189  
57                        10.5              0.041063  
55                         1.8              0.038280  
53                         0.8              0.038249  
94                         5.7              0.037958  
84                         4.2              0.035056  
66                         9.4              0.032811  
96                         0.4              0.032174  
0                         10.5              0.028929  
24                         1.4              0.027394  
102                        8.0              0.027209  
89                         2.0              0.026708  
83                        13.4              0.024515  
47                         9.0              0.023966  
50                        11.9              0.023695  
13                         0.5              0.022675  
97                         1.6              0.020783  
26                         4.9              0.020051  
33                        21.1              0.019831

## **6.2 Les pays qui ont le plus bénéficié d’aide depuis 2013**

In [ ]:
aide_alimentaire_2013_2017=(aide_alimentaire[(aide_alimentaire['Année']>=2013)])
aide_alimentaire_2013_2017.groupby(by='Pays bénéficiaire').agg({'Valeur':sum}).sort_values(by=['Valeur'], ascending=False)


Valeur
Pays bénéficiaire                       
République arabe syrienne        1858943
Éthiopie                         1381294
Yémen                            1206484
Soudan du Sud                     695248
Soudan                            669784
...                                  ...
Égypte                              1122
Vanuatu                              802
Timor-Leste                          116
Géorgie                               70
Bolivie (État plurinational de)        6

[76 rows x 1 columns]

## **6.3 Les pays qui ont le plus/le moins de disponibilité alimentaire par habitant**

### 6.3.1 Considerant la **disponibilité intérieure**

In [ ]:
#les pays qui ont le plus de disponibilité alimentaire par habitant
dispo_int_zone=dispo_alimentaire.groupby(by='Zone').agg({'Disponibilité intérieure':sum})
dispo_int_zone=dispo_int_zone.sort_values(['Disponibilité intérieure'], ascending=False)
dispo_int_zone.head(20)

Disponibilité intérieure
Zone                                           
Chine, continentale                   2036726.0
Inde                                  1100838.0
Brésil                                1033314.0
États-Unis d'Amérique                  777992.0
Fédération de Russie                   263784.0
Indonésie                              226930.0
Nigéria                                192388.0
Mexique                                186709.0
Pakistan                               168631.0
Allemagne                              162275.0
Thaïlande                              158130.0
France                                 146905.0
Turquie                                131377.0
Égypte                                 115133.0
Argentine                              114369.0
Ukraine                                113924.0
Japon                                  113471.0
Philippines                            108885.0
Espagne                                101484.0
Italie                                  97405.0

In [ ]:
merge_dispo_int_zone=dispo_alimentaire_pop_2017.merge(dispo_int_zone, on = ['Zone'])
merge_dispo_int_zone

Zone Année     Valeur  \
0                               Afghanistan  2017  36296.113   
1                            Afrique du Sud  2017  57009.756   
2                                   Albanie  2017   2884.169   
3                                   Algérie  2017  41389.189   
4                                 Allemagne  2017  82658.409   
..                                      ...   ...        ...   
167  Venezuela (République bolivarienne du)  2017  29402.484   
168                                Viet Nam  2017  94600.648   
169                                   Yémen  2017  27834.819   
170                                  Zambie  2017  16853.599   
171                                Zimbabwe  2017  14236.595   

     Disponibilité intérieure  
0                     13515.0  
1                     61256.0  
2                      4758.0  
3                     42630.0  
4                    162275.0  
..                        ...  
167                   32287.0  
168                   95134.0  
169                    9782.0  
170                   10041.0  
171                    9507.0  

[172 rows x 4 columns]

6.3.2 Considerant **la disponibilité alimentaire (kcal/personne/jour)**

In [ ]:
#disponibilité alimentaire en calorie par habitant
#les pays qui ont le plus de disponibilité alimentaire par habitant
dispo_zone=dispo_alimentaire.groupby(by='Zone').agg({'Disponibilité alimentaire (Kcal/personne/jour)':sum})
dispo_zone=dispo_zone.sort_values(['Disponibilité alimentaire (Kcal/personne/jour)'], ascending=False)
dispo_zone.head(20)

Disponibilité alimentaire (Kcal/personne/jour)
Zone                                                                 
Autriche                                                       3770.0
Belgique                                                       3737.0
Turquie                                                        3708.0
États-Unis d'Amérique                                          3682.0
Israël                                                         3610.0
Irlande                                                        3602.0
Italie                                                         3578.0
Luxembourg                                                     3540.0
Égypte                                                         3518.0
Allemagne                                                      3503.0
Koweït                                                         3499.0
Canada                                                         3499.0
Monténégro                                                     3490.0
Norvège                                                        3483.0
France                                                         3482.0
Portugal                                                       3472.0
Pologne                                                        3450.0
Royaume-Uni                                                    3425.0
Lituanie                                                       3414.0
Cuba                                                           3402.0

In [ ]:
#disponibilité alimentaire en calorie par habitant
#les pays qui ont le moins de disponibilité alimentaire par habitant
dispo_zone=dispo_alimentaire.groupby(by='Zone').agg({'Disponibilité alimentaire (Kcal/personne/jour)':sum})
dispo_zone=dispo_zone.sort_values(['Disponibilité alimentaire (Kcal/personne/jour)'], ascending=True)
dispo_zone.head(20)

Disponibilité alimentaire (Kcal/personne/jour)
Zone                                                                                      
République centrafricaine                                                           1879.0
Zambie                                                                              1924.0
Madagascar                                                                          2056.0
Afghanistan                                                                         2087.0
Haïti                                                                               2089.0
République populaire démocratique de Corée                                          2093.0
Tchad                                                                               2109.0
Zimbabwe                                                                            2113.0
Ouganda                                                                             2126.0
Timor-Leste                                                                         2129.0
Éthiopie                                                                            2129.0
Namibie                                                                             2166.0
Tadjikistan                                                                         2199.0
République-Unie de Tanzanie                                                         2204.0
Kenya                                                                               2205.0
Libéria                                                                             2206.0
Congo                                                                               2206.0
Yémen                                                                               2217.0
Rwanda                                                                              2224.0
Bolivie (État plurinational de)                                                     2250.0

### 6.3.1 Considerant la **disponibilité alimentaire en calories**

In [ ]:
dispo_alimentaire

In [ ]:
merge_dispo_int_zone['Ratio population et disp. alimentaire']=merge_dispo_int_zone['Valeur']/merge_dispo_int_zone['Disponibilité intérieure']
merge_dispo_int_zone=merge_dispo_int_zone.sort_values(['Ratio population et disp. alimentaire'], ascending=True)
merge_dispo_int_zone.head(20)

Zone Année      Valeur  Disponibilité intérieure  \
53                Eswatini  2017    1124.805                    6467.0   
24                  Brésil  2017  207833.823                 1033314.0   
69                  Guyana  2017     775.222                    3290.0   
18                  Belize  2017     375.769                    1542.0   
104                Maurice  2017    1264.499                    4767.0   
44                Danemark  2017    5732.274                   19749.0   
16                 Bélarus  2017    9450.231                   27522.0   
57                   Fidji  2017     877.459                    2499.0   
10               Australie  2017   24584.620                   66992.0   
94                Lituanie  2017    2845.414                    7467.0   
8                Argentine  2017   43937.140                  114369.0   
17                Belgique  2017   11419.748                   29358.0   
164                Ukraine  2017   44487.709                  113924.0   
43                    Cuba  2017   11339.254                   28681.0   
125               Pays-Bas  2017   17021.347                   42461.0   
124               Paraguay  2017    6867.061                   17016.0   
166                Vanuatu  2017     285.510                     700.0   
54   États-Unis d'Amérique  2017  325084.756                  777992.0   
157              Thaïlande  2017   69209.810                  158130.0   
59                  France  2017   64842.509                  146905.0   

     Ratio population et disp. alimentaire  
53                                0.173930  
24                                0.201133  
69                                0.235630  
18                                0.243689  
104                               0.265261  
44                                0.290256  
16                                0.343370  
57                                0.351124  
10                                0.366978  
94                                0.381065  
8                                 0.384170  
17                                0.388982  
164                               0.390503  
43                                0.395358  
125                               0.400870  
124                               0.403565  
166                               0.407871  
54                                0.417851  
157                               0.437677  
59                                0.441391

In [ ]:
merge_dispo_int_zone['Ratio population et disp. alimentaire']=merge_dispo_int_zone['Valeur']/merge_dispo_int_zone['Disponibilité intérieure']
merge_dispo_int_zone=merge_dispo_int_zone.sort_values(['Ratio population et disp. alimentaire'], ascending=False)
merge_dispo_int_zone.head(20)

Zone Année      Valeur  \
169                                       Yémen  2017   27834.819   
156                                       Tchad  2017   15016.753   
158                                 Timor-Leste  2017    1243.258   
0                                   Afghanistan  2017   36296.113   
55                                     Éthiopie  2017  106399.924   
61                                       Gambie  2017    2213.889   
93                                      Libéria  2017    4702.226   
68                                Guinée-Bissau  2017    1828.145   
14                                   Bangladesh  2017  159685.424   
90                                      Lesotho  2017    2091.534   
145                                     Sénégal  2017   15419.355   
105                                  Mauritanie  2017    4282.570   
114                                       Niger  2017   21602.382   
159                                        Togo  2017    7698.474   
45                                     Djibouti  2017     944.099   
131                   République centrafricaine  2017    4596.023   
70                                        Haïti  2017   10982.366   
23                                     Botswana  2017    2205.080   
77                                         Iraq  2017   37552.781   
136  République populaire démocratique de Corée  2017   25429.825   

     Disponibilité intérieure  Ratio population et disp. alimentaire  
169                    9782.0                               2.845514  
156                    5304.0                               2.831213  
158                     447.0                               2.781338  
0                     13515.0                               2.685617  
55                    44014.0                               2.417411  
61                      940.0                               2.355201  
93                     2002.0                               2.348764  
68                      786.0                               2.325884  
14                    72757.0                               2.194777  
90                      966.0                               2.165149  
145                    7495.0                               2.057286  
105                    2142.0                               1.999332  
114                   10997.0                               1.964389  
159                    4095.0                               1.879969  
45                      526.0                               1.794865  
131                    2582.0                               1.780024  
70                     6175.0                               1.778521  
23                     1253.0                               1.759840  
77                    21723.0                               1.728711  
136                   14876.0                               1.709453